File: backend/Car Details.py
from flask import Flask, jsonify, request
import time

app = Flask(__name__)

# Mock data for car details
car_data = {
    '1': {'make': 'Toyota', 'model': 'Camry', 'year': 2020, 'color': 'Blue'},
    '2': {'make': 'Honda', 'model': 'Accord', 'year': 2021, 'color': 'Red'},
    '3': {'make': 'Ford', 'model': 'Mustang', 'year': 2019, 'color': 'Black'}
}

@app.route('/car/<carId>', methods=['GET'])
def get_car_details(carId):
    start_time = time.time()
    car_detail = car_data.get(carId)
    if car_detail:
        response = jsonify(car_detail)
    else:
        response = jsonify({'error': 'Car not found'}), 404
    elapsed_time = (time.time() - start_time) * 1000
    if elapsed_time > 200:
        print('Warning: Response time exceeded 200ms')
    return response

if __name__ == '__main__':
    app.run(debug=True, port=5000)

import React, { useEffect, useState } from 'react';

const CarDetails = ({ carId }) => {
    const [carDetail, setCarDetail] = useState(null);
    const [error, setError] = useState(null);

    useEffect(() => {
        const fetchCarDetails = async () => {
            try {
                const response = await fetch(`http://localhost:5000/car/${carId}`);
                if (!response.ok) {
                    throw new Error('Car not found');
                }
                const data = await response.json();
                setCarDetail(data);
            } catch (err) {
                setError(err.message);
            }
        };
        fetchCarDetails();
    }, [carId]);

    if (error) {
        return <div>{error}</div>;
    }

    if (!carDetail) {
        return <div>Loading...</div>;
    }

    return (
        <div>
            <h1>Car Details</h1>
            <p>Make: {carDetail.make}</p>
            <p>Model: {carDetail.model}</p>
            <p>Year: {carDetail.year}</p>
            <p>Color: {carDetail.color}</p>
        </div>
    );
};

export default CarDetails;

File: frontend/src/Car Details.js
from flask import Flask, jsonify, request

app = Flask(__name__)

# Sample car data
cars = {
    '1': {'name': 'Car A', 'image': 'url_to_image_A', 'specs': {'color': 'red', 'engine': 'V6', 'year': 2020}},
    '2': {'name': 'Car B', 'image': 'url_to_image_B', 'specs': {'color': 'blue', 'engine': 'V8', 'year': 2021}}
}

@app.route('/api/cars/<car_id>', methods=['GET'])
def get_car_details(car_id):
    car_detail = cars.get(car_id)
    if car_detail:
        return jsonify(car_detail), 200
    return jsonify({'error': 'Car not found'}), 404

if __name__ == '__main__':
    app.run(debug=True)

import React, { useEffect, useState } from 'react';

const CarDetails = ({ carId }) => {
    const [carDetail, setCarDetail] = useState(null);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        const fetchCarDetails = async () => {
            const response = await fetch(`/api/cars/${carId}`);
            if (response.ok) {
                const data = await response.json();
                setCarDetail(data);
            }
            setLoading(false);
        };
        fetchCarDetails();
    }, [carId]);

    if (loading) return <div>Loading...</div>;
    if (!carDetail) return <div>Car not found</div>;

    return (
        <div>
            <h1>{carDetail.name}</h1>
            <img src={carDetail.image} alt={carDetail.name} />
            <h2>Specifications:</h2>
            <ul>
                <li>Color: {carDetail.specs.color}</li>
                <li>Engine: {carDetail.specs.engine}</li>
                <li>Year: {carDetail.specs.year}</li>
            </ul>
        </div>
    );
};

export default CarDetails;